crates/transaction/manifests/src/manifests_security_shield/lock_fee_against_xrd_vault_of_access_controller.rs

```diff
@@ -0,0 +1,52 @@
+use profile_supporting_types::AnySecurifiedEntity;
+use radix_engine_interface::blueprints::access_controller::{
+    AccessControllerLockRecoveryFeeInput as ScryptoAccessControllerLockRecoveryFeeInput,
+    ACCESS_CONTROLLER_LOCK_RECOVERY_FEE_IDENT as SCRYPTO_ACCESS_CONTROLLER_LOCK_RECOVERY_FEE_IDENT,
+};
+use radix_transactions::prelude::ManifestBuilder;
+
+use crate::prelude::*;
+
+pub trait TransactionManifestLockFeeAgainstXrdVaultOfAccessController {
+    /// Locks transaction fee against the XRD vault of the access controller of
+    /// `entity_applying_shield` - either AC of an Account or of a Persona.
+    ///
+    /// We need to call this later when we have made a preview/dry-run of the
+    /// `manifest` to get the actual fee to lock.
+    ///
+    /// `manifest` was produced by `apply_security_shield_for_securified_entity`.
+    ///
+    /// In fact will will be locking fee for 6 flavours of transaction manifest
+    /// which updates the security shield of an entity, as returned by
+    /// `TransactionManifestApplySecurityShieldKind::all()`, and we could try to
+    /// be smart and run preview of each six to get a total fee to lock, but we
+    /// will not do that, we will just lock the fee for the biggest one. Since
+    /// only the required amount is drawn.
+    ///
+    /// Only relevant for securified entities - since it is only securified entities
+    /// which have an access controller to lock against.
+    fn modify_manifest_add_lock_fee_against_xrd_vault_of_access_controller(
+        manifest: TransactionManifest,
+        fee: Decimal192,
+        // TODO: remove `entity_applying_shield`, this should be read out from the manifest in a throwing function, `manifest.get_address_of_entity_applying_shield()` or similar which Omar need to provide us with, oh well we need the account here, so elsewhere, in SargonOS where we have access to Profile we would call `manifest.get_address_of_entity_applying_shield` and then lookup the entity.
+        entity_applying_shield: AnySecurifiedEntity,
+    ) -> TransactionManifest {
+        let mut builder = ManifestBuilder::with_manifest(manifest);
+        let access_controller_address = entity_applying_shield
+            .securified_entity_control
+            .access_controller_address;
+        // Lock fee against XRD vault of the access controller
+        builder = builder.call_method(
```

Comment:
@0xOmarA here I'm appending this instruction **last**. Is that OK? Or must this be **prepended**, index 0 of the manifest? if so I need to do a lot of trickery to do that. Since I cannot create the instruction in rust code "manually" and since calling `buider.call_method` will put it last...

@GhenadieVP alternatively we just lock a large amount. Which ""should work"" for most shields... and if we do that we can lock fee in `apply_security_shield_for_securified_entity` at index 0  https://github.com/radixdlt/sargon/blob/ac/wallet_interaction_applying_shield_abw-4056_part2/crates/transaction/manifests/src/manifests_security_shield/manifests_securify_shield_securified_entity.rs#L37